develop/networking.md

```diff
@@ -0,0 +1,252 @@
+---
+title: Networking
+description: A general guide on networking using Fabric API.
+authors:
+  - Daomephsta
+  - dicedpixels
+  - Earthcomputer
+  - FlooferLand
+  - FxMorin
+  - i509VCB
+  - modmuss50
+  - natanfudge
+  - NetUserGet
+  - NShak
+  - parzivail
+  - skycatminepokie
+  - SolidBlock-cn
+  - Voleil
+  - Wxffel
+  - YTG123-Mods
+  - zulrang
+---
+
+Networking in Minecraft is used so the client and server can communicate with each other. Networking is a broad topic,
+so this page is split up into a few categories.
+
+## Why Is Networking Important? {#why-is-networking-important}
+
+The importance of networking can be shown by a simple code example.
+
+::: warning
+Below code is for demonstration purposes only.
+:::
+
+Say you had a Wand which highlights the block you're looking, which will be visible to all
+nearby players:
+
+```java
+public class HighlightingWandItem extends Item {
+    public HighlightingWandItem(Item.Settings settings) {
+        super(settings);
+    }
+
+    public TypedActionResult<ItemStack> use(World world, PlayerEntity user, Hand hand) {
+        BlockPos target = ...
+
+        // BAD CODE: DON'T EVER DO THIS! // [!code error]
+        ClientBlockHighlighting.highlightBlock(MinecraftClient.getInstance(), target); // [!code error]
+        return super.use(world, user, hand);
+    }
+}
+```
+
+Upon testing, you will see a lightning bolt being summoned and nothing crashes. Now you want to show the mod to your
+friend, you boot up a dedicated server and invite your friend on with the mod installed.
+
+You use the item and the server crashes. You will probably notice in the crash log an error similar to this:
+
+```log
+[Server thread/FATAL]: Error executing task on Server
+java.lang.RuntimeException: Cannot load class net.minecraft.client.MinecraftClient in environment type SERVER
+```
+
+### Why Does the Server Crash? {#why-does-the-server-crash}
+
+The code calls logic only present on the client distribution of the Minecraft. The reason for Mojang distributing the
+game in this way is to cut down on the size of the Minecraft Server JAR file. There isn't really a reason to include an
+entire rendering engine when your own machine will render the world.
+
+In a development environment, client-only classes are indicated by the `@Environment(EnvType.CLIENT)` annotation.
+
+### How Do We Fix the Crash? {#how-do-we-fix-the-crash}
+
+To fix this issue, you need to understand how the game client and dedicated server communicate:
+
+![Sides](/assets/develop/networking/sides.png)
+
+The diagram above shows that the game client and dedicated server are separate systems, bridged together using
+_packets_. Packets can contain data which we refer to as the _payload_.
```

Comment:
This is fine I guess, it just doenst have anything to do with networking. Packets are still sent between the logicial client and server even in single player, or when on a LAN world. A better example would be to cause desync by setting something only the logicial client. Maybe try adding a block to the hotbar from a keybind or something.